### PR TITLE
Adds spans to v2 tests

### PR DESCRIPTION
**Pull request by therve**
This adds spans to v2 tests.

### CODE REVIEW

File: tests/test_utils.go [hunk 1]
```diff
@@ -354,6 +354,7 @@ func ClockFromContext(ctx context.Context) clockwork.FakeClock {
 	v := ctx.Value(clockKey)
 	fc, ok := v.(clockwork.FakeClock)
 	if !ok {
+		panic("woops")
```
Comment (nmuesch) [inline tests/test_utils.go:357]: leftover? 😉

File: tests/api/v2/datadog/api_dashboard_lists_test.go [hunk 2]
```diff
@@ -65,7 +65,9 @@ func deleteDashboardList() {
 }
 
 func TestDashboardListItemCRUD(t *testing.T) {
-	ctx, finish := WithRecorder(WithTestAuth(context.Background()), t)
+	ctx, finish := tests.WithTestSpan(context.Background(), t)
+	defer finish()
+	ctx, finish = WithRecorder(WithTestAuth(ctx), t)
 	defer finish()
```
Comment (nmuesch) [inline tests/api/v2/datadog/api_dashboard_lists_test.go:71]: Not super familiar, but does the double `defer` with the same method name work as expected here?

File: tests/test_utils.go (hunk 1 above)
Comment (therve) [reply tests/test_utils.go]: Thanks :).

File: tests/api/v2/datadog/api_dashboard_lists_test.go (hunk 2 above)
Comment (therve) [reply tests/api/v2/datadog/api_dashboard_lists_test.go:71]: AFAIK there is no name binding, so it's not a problem. At least recording works fine, we can check if spans are reported properly.

Comment (nmuesch) [reply tests/api/v2/datadog/api_dashboard_lists_test.go:71]: Ok 👍 Thanks

**Review by nmuesch — APPROVED** [review]

**Review by jirikuncar — APPROVED** [review]

**Pull request merged**